junit-platform-testkit/src/main/java/org/junit/platform/testkit/engine/EngineTestKit.java

```diff
@@ -249,8 +250,8 @@ private static void executeDirectly(TestEngine testEngine, EngineDiscoveryReques
 			EngineExecutionListener listener) {
 		UniqueId engineUniqueId = UniqueId.forEngine(testEngine.getId());
 		TestDescriptor engineTestDescriptor = testEngine.discover(discoveryRequest, engineUniqueId);
-		ExecutionRequest request = new ExecutionRequest(engineTestDescriptor, listener,
-			discoveryRequest.getConfigurationParameters());
+		ExecutionRequest request = ExecutionRequest.create(engineTestDescriptor, listener,
+			discoveryRequest.getConfigurationParameters(), OutputDirProvider.NOOP);
```

Comment:
For testing with `EngineTestKit` a special implementation of `OutputDirProvider` should be provided that uses a temporary directory that is cleaned up after the surrounding test is finished.